frontend/packages/kubevirt-plugin/src/components/vms/pending-changes-warning.tsx

```diff
@@ -0,0 +1,136 @@
+import { VMWrapper } from '../../k8s/wrapper/vm/vm-wrapper';
+import { VMIWrapper } from '../../k8s/wrapper/vm/vmi-wrapper';
+import { PendingChangesAlert } from '../Alerts/PendingChangesAlert';
+import {
+  List,
+  ListItem,
+  Button,
+  Breadcrumb,
+  BreadcrumbItem,
+  BreadcrumbHeading,
+  ButtonVariant,
+} from '@patternfly/react-core';
+import { asVM } from '../../selectors/vm';
+import { VirtualMachineModel, VirtualMachineInstanceModel } from '../../models';
+import { PENDING_CHANGES_WARNING_MESSAGE } from '../../strings/vm/status';
+import { isVMRunningOrExpectedRunning } from '../../selectors/vm/selectors';
+import { getPendingChanges, hasPendingChanges } from '../../utils/pending-changes';
+
+import './pending-changes-warning.scss';
+
+type PendingChangesWarningProps = {
+  vmi?: FirehoseResult<VMIKind>;
+  vmLikeEntity?: FirehoseResult<VMLikeEntityKind>;
+};
+
+const getPendingChangesByTab = (pendingChanges: PendingChanges) =>
+  Object.keys(pendingChanges)
+    .filter((key) => pendingChanges[key].isPendingChange)
+    .reduce((acc, key) => {
+      const pc = pendingChanges[key];
+      return {
+        ...acc,
+        [pc.vmTab]: acc[pc.vmTab] ? [...acc[pc.vmTab], key] : [key],
+      };
+    }, {});
+
+type WarningTabRowProps = {
+  tabName: string;
+  tabProps: string[];
+  pendingChanges: PendingChanges;
+};
+
+const WarningTabRow: React.FC<WarningTabRowProps> = ({ tabName, tabProps, pendingChanges }) => (
+  <Breadcrumb className="kv-warning--pf-c-breadcrumb">
+    <BreadcrumbHeading key={`${tabName}-header-${tabProps.join('-')}`}>{tabName}</BreadcrumbHeading>
+    <BreadcrumbItem key={`${tabName}-${tabProps.join('-')}`}>
+      {tabProps.map((key, idx) => (
+        <div key={`${tabName}-${key}`} className="kv-warning-changed-attrs">
+          <Button isInline onClick={pendingChanges[key].execAction} variant={ButtonVariant.link}>
+            {key}
+          </Button>
+          <div className="kv-warning--comma-margin">{idx === tabProps.length - 1 ? '' : ','}</div>
+        </div>
+      ))}
+    </BreadcrumbItem>
+  </Breadcrumb>
+);
+
+export const PendingChangesWarning: React.FC<PendingChangesWarningProps> = ({
+  vmLikeEntity,
+  vmi: vmiProp,
+}) => {
+  const vm: VMKind = asVM(getLoadedData(vmLikeEntity));
+  if (!isVMRunningOrExpectedRunning(vm)) {
+    return <></>;
+  }
+
+  const vmi = getLoadedData(vmiProp);
+
+  const vmWrapper = new VMWrapper(vm);
+  const vmiWrapper = new VMIWrapper(vmi);
+
+  const pendingChanges = getPendingChanges(vmWrapper, vmiWrapper);
+  const arePendingChanges = hasPendingChanges(vm, vmi, pendingChanges);
+  const pendingChangesByTab = getPendingChangesByTab(pendingChanges);
+
+  if (Object.keys(pendingChanges).length === 0 || !arePendingChanges) {
+    return <></>;
+  }
+
+  return (
+    <PendingChangesAlert isWarning>
+      {PENDING_CHANGES_WARNING_MESSAGE}
+      <List className="kv-warning--pf-c-list">
+        {Object.keys(pendingChangesByTab).map(
```

Comment:
```suggestion
        {Object.keys(getPendingChangesByTab(pendingChanges)).map(
```

IMO better to move here, since we do not need the variable before early return

Reply:
I'm using `pendingChangesByTab` later too. 
moving the variable creation after the early return.

Reply:
+1